qiskit/primitives/containers/observables_array.py

```diff
@@ -273,6 +281,73 @@ def coerce(cls, observables: ObservablesArrayLike) -> ObservablesArray:
             return observables
         return cls(observables)
 
+    def equivalent(self, other: ObservablesArray, tol: float = 1e-08):
+        """Return True if arrays are equal"""
+        if self.num_qubits != other.num_qubits or self.shape != other.shape:
+            return False
+
+        arr1 = np.atleast_1d(self._array).flat
+        arr2 = np.atleast_1d(other._array).flat
+
+        for obs1, obs2 in zip(arr1, arr2):
+            if (obs1 - obs2).simplify(tol) != SparseObservable.zero(self.num_qubits):
+                return False
+
+        return True
+
+    def copy(self):
+        """Return a deep copy of the array"""
+        return deepcopy(self)
+
+    def apply_layout(
+        self, layout: TranspileLayout | List[int] | None, num_qubits: int | None = None
+    ) -> ObservablesArray:
+        """Apply a transpiler layout to this :class:`~.ObservablesArray`
+
+        Args:
+            layout: Either a :class:`~.TranspileLayout`, a list of integers or None.
+                    If both layout and num_qubits are none, a deep copy of the array is
+                    returned.
+            num_qubits: The number of qubits to expand the array to. If not
+                provided then if ``layout`` is a :class:`~.TranspileLayout` the
+                number of the transpiler output circuit qubits will be used by
+                default. If ``layout`` is a list of integers the permutation
+                specified will be applied without any expansion. If layout is
+                None, the array will be expanded to the given number of qubits.
+
+        Returns:
+            A new :class:`.ObservablesArray` with the provided layout applied
+        """
+        from qiskit.transpiler.layout import TranspileLayout
+
+        if layout is None and num_qubits is None:
+            return self.copy()
+
+        n_qubits = self.num_qubits
+        if isinstance(layout, TranspileLayout):
+            n_qubits = len(layout._output_qubit_list)
```

Comment:
We shouldn't be using private attributes of other classes. Is there another way to fetch this information?

Reply:
It seems like just `len(layout)` is sufficient?